src/theory/arith/nl/ext/monomial_check.h

```diff
@@ -26,7 +26,7 @@ namespace nl {
 class MonomialCheck
 {
  public:
-  MonomialCheck(ExtState* data);
+  MonomialCheck(ExtState* data, context::Context* ctx);
```

Comment:
Can you call `ctx` -> `u` ?  The point is it should be clear that this is the user context. (Although this class could be a generic utility, its more important to me to make its standard use case clear.)